src/Sarif/NotYetAutoGenerated/Result.cs

```diff
@@ -83,22 +84,47 @@ public ResultKind Kind
             }
         }
 
-        private FailureLevel _level;
+        private FailureLevel? _level;
 
         /// <summary>
         /// A value specifying the severity level of the result.
         /// </summary>
         [DataMember(Name = "level", IsRequired = false, EmitDefaultValue = false)]
-        [DefaultValue(FailureLevel.Warning)]
-        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
         [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.EnumConverter))]
-        public FailureLevel Level
+        public FailureLevel? Level
         {
-            get { return _level; }
+            get
+            {
+                if (_level.HasValue || this.Run == null || this.Run.GetToolComponentFromReference(this.Rule?.ToolComponent)?.Rules == null)
+                {
+                    return _level;
+                }
+                else
+                {
+                    ReportingConfiguration currentDefaultConfiguration = this.GetRule()?.DefaultConfiguration;
+
+                    if (currentDefaultConfiguration == null)
+                    {
+                        // if defaultConfiguration is absent, it SHALL be taken to be present and default Warning
+                        return FailureLevel.Warning;
+                    }
+                    else if (!currentDefaultConfiguration.Enabled)
+                    {
+                        // if defaultConfiguration is present, but disabled
+                        return _level;
```

Comment:
this is my understanding of the spec, let me know if we want to have this condition.

Reply:
According to spec
"IF theDescriptor.defaultConfiguration.level (§3.49.14, §, §3.50.3) is present THEN
          SET level to theDescriptor.defaultConfiguration.level.
IF level has not yet been set THEN
    SET level to "warning"."

i think it may be similar to:
if (currentDefaultConfiguration != null && currentDefaultConfiguration.Enable)
{
    _level = currentDefaultConfiguration.Level;
}
return _level ?? FailureLevel.Warning;

Reply:
Thanks, you are correct. I have moved this code to extension method.